File: lib/lib-kotlin/src/main/java/com/jay/lib_kotlin/delegate/Person.java
package com.jay.lib_kotlin.delegate;

import java.lang.reflect.Field;

/**
 * java 实现 kotlin 属性委托
 *
 * @author jaydroid
 * @version 1.0
 * @date 2021/9/9
 */
class Person {
  static final Field[] delegatedProperties = Person.class.getFields();
  private final NameDelegate nameDelegate = new NameDelegate();
  public final String getName() {
    return this.nameDelegate.getValue(this, delegatedProperties[0]);
  }
}
class NameDelegate {
  String getValue(Person thisRef, Field property) {
    return "Jay";
  }
}

File: lib/lib-kotlin/src/main/java/com/jay/lib_kotlin/delegate/Person.kt
package com.jay.lib_kotlin.delegate

/**
 * @author jaydroid
 * @version 1.0
 * @date 2021/9/2
 */
class Person(name: String) {

    val a: String = name.split(" ")[0]

    //public actual fun <T> lazy(initializer: () -> T): Lazy<T> = SynchronizedLazyImpl(initializer)
    val firstName: String by lazy {
        a
    }

}